src/main/java/io/pravega/connectors/flink/FlinkPravegaInputFormat.java

```diff
@@ -137,7 +137,7 @@ public BaseStatistics getStatistics(BaseStatistics cachedStatistics) throws IOEx
                 for (Iterator<SegmentInfo> segmentInfos = batchClient.listSegments(new StreamImpl(scopeName, stream)); segmentInfos.hasNext(); ) {
                     SegmentInfo segmentInfo = segmentInfos.next();
                     Segment segment = segmentInfo.getSegment();
-                    splits.add(new PravegaInputSplit(splits.size(), segment, 0, segmentInfo.getLength()));
+                    splits.add(new PravegaInputSplit(splits.size(), segment, segmentInfo.getStartingOffset(), segmentInfo.getWriteOffset() - 1));
```

Comment:
Turns out that the call to `batchClient::readSegment` has an exclusive `endingOffset`.   But the `PravegaInputSplit` is written to be inclusive.  So we need to change the call to `readSegment` below, or change the `PravegaInputSplit`.

cc @tkaitchuck

Reply:
@EronWright I'm not sure I follow why do you say that `PravegaInputSplit` is written to be inclusive. There is no description mentioning it, although there is this precondition, which is perhaps what is making you say it:

https://github.com/pravega/flink-connectors/blob/master/src/main/java/io/pravega/connectors/flink/PravegaInputSplit.java#L36

If there is any reason for it to be inclusive, then I think we do need to subtract and add, but it is not clear that it is needed. Could you clarify, please?

Reply:
There isn't any specific requirement for the `PravegaInputSplit` to have an inclusive end offset.
Though, it would make sense to clarify this in the Javadocs.

Reply:
Sorry for the confusion. I am removing this parameter to avoid this in the future: https://github.com/pravega/pravega/pull/2216

Reply:
I agree that the split could easily have an exclusive `endOffset`, and that its javadocs should be updated in any case.   My only objection is that the name `endOffset` suggests inclusiveness (to me).   I can get over that.